src/main/java/timify/com/studytime/StudyTimeService.java

```diff
@@ -0,0 +1,232 @@
+                boundaryDateTime.format(DateTimeFormatter.ofPattern("yyyyMMddHHmmss")),
+                request.getGrade()),
+            calculateTemp(todo, startTime, boundaryDateTime, request.getGrade()));
+        studyTime.associateMember(member);
+        studyTime.associateSubject(todo.getSubject());
+        studyTime.associateTodo(todo);
+        return studyTimeRepository.save(studyTime);
+    }
+
+    private StudyTime saveStudyTimePartForNextDay(Member member, Todo todo,
+        LocalDateTime boundaryDateTime, LocalDateTime endTime, studyTimeRequest request) {
+
+        Todo nextDayTodo = findOrCreateNextDayTodo(todo,
+            boundaryDateTime.toLocalDate().plusDays(1));
+        StudyTime studyTime = StudyTimeConverter.toStudyTime(new studyTimeRequest(
+                boundaryDateTime.format(DateTimeFormatter.ofPattern("yyyyMMddHHmmss")),
+                request.getEndTime(), request.getGrade()),
+            calculateTemp(nextDayTodo, boundaryDateTime, endTime, request.getGrade()));
+
+        studyTime.associateMember(member);
+        studyTime.associateSubject(nextDayTodo.getSubject());
+        studyTime.associateTodo(nextDayTodo);
+        return studyTimeRepository.save(studyTime);
+    }
+
+    private StudyTime saveStudyTime(Member member, Todo todo, LocalDateTime startTime,
+        LocalDateTime endTime, studyTimeRequest request) {
+        StudyTime studyTime = StudyTimeConverter.toStudyTime(request,
+            calculateTemp(todo, startTime, endTime, request.getGrade()));
+
+        studyTime.associateMember(member);
+        studyTime.associateSubject(todo.getSubject());
+        studyTime.associateTodo(todo);
+        return studyTimeRepository.save(studyTime);
+    }
+
+    private Todo findOrCreateNextDayTodo(Todo todo, LocalDate nextDayDate) {
+        return todoRepository.findByMemberAndSubjectId(todo.getMember(), todo.getSubject().getId())
+            .stream().filter(existingTodo -> existingTodo.getDate().equals(nextDayDate)).findFirst()
+            .orElseGet(() -> {
+                Todo newTodo = TodoConverter.toTodo(new todoRequest(todo.getContent(), nextDayDate,
+                        todo.getStudyType() != null ? todo.getStudyType().getId() : null,
+                        todo.getStudyMethod() != null ? todo.getStudyMethod().getId() : null,
+                        todo.getStudyPlace() != null ? todo.getStudyPlace().getId() : null),
+                    todo.getStudyType(), todo.getStudyMethod(), todo.getStudyPlace());
+
+                newTodo.associateMember(todo.getMember());
+                newTodo.associateSubject(todo.getSubject());
+                return todoRepository.save(newTodo);
+            });
+    }
+
+    private double calculateTemp(Todo todo, LocalDateTime startTime, LocalDateTime endTime,
+        StudyTimeGrade grade) {
+
+        int minutes = (int) Duration.between(startTime, endTime).toMinutes();
+        long countStudyTime = studyTimeRepository.countByTodo(todo);
```

Comment:
repository를 쓸 필요 없이, todo 엔티티에 studyTimeList를 가지고 있으니 해당 list의 size를 구하면 될 것 같아요!

```suggestion
        long countStudyTime = todo.getStudyTimeList().size();
```